docs/architecture/0004-eq-author-permissions-model.md

```diff
@@ -0,0 +1,64 @@
+# 4. EQ Author Permissions Model
+
+Date: 2019-03-11
+
+## Status
+
+Accepted
+
+## Context
+
+As part of the work to move Author into a live environment, in order to do that we have business requirements to fulfill around that to introduce controls around who can view and modify questionnaires in the system. Currently this is not locked down in any way, leading to undesired consequences, people can overwrite someone else's work and there's no clear audit trail at the moment about who made the change and what change was made.
+
+We need to solve these problems, and the simplest way we can begin to solve some of these problems is to introduce permissions based model to Author.
+
+## Decision
+
+We're proposing to implement a permissions model for Author with the following design.
+
+### Querying a questionnaire
+
+There are two parts to consider with this:
+
+1. Who has access to see the questionnaire.
+1. Who has access to make changes to a questionnaire.
+
+When the user logs in:
+
+1. Get the current user Id from the token.
+1. Loop through the questionnaire list.
+1. For each questionnaire check that the user has questionnaire read permission by looking it up in questionnaire list table.
+1. Show the questionnaire in the list if the user has read permissions, if the user doesn't have read permissions do not show questionnaire in the list and restrict them from being able to load the questionnaire.
```

Comment:
I wouldn't assume this at this point. It might be that we provide the users a way to filter questionnaires based on their permissions rather than not returning them at all.

In this scenario, they could still see the questionnaire in the list but not access it.